lib/SimpleSAML/Logger.php

```diff
@@ -489,15 +489,11 @@ private static function log($level, $string, $statsLog = false)
         if (self::$captureLog) {
             $ts = microtime(true);
             $msecs = (int) (($ts - (int) $ts) * 1000);
-            $ts = gmdate('H:i:s', $ts) . sprintf('.%03d', $msecs) . 'Z';
+            $ts = gmdate('H:i:s') . sprintf('.%03d', $msecs) . 'Z';
```

Comment:
`<?php

$ts = floatval(strval(time()).'.999');
$msecs = (int) (($ts - (int) $ts) * 1000);

$ts = gmdate('H:i:s', $ts) . sprintf('.%03d', $msecs) . 'Z';
echo "$ts\n";

$ts = gmdate('H:i:s') . sprintf('.%03d', $msecs) . 'Z';
echo "$ts\n";
`

I cannot reproduce the case you describe here, or am I doing it wrong?  Anyway, param2 for `gmdate()` must be int instead of float...

Reply:
The problem is that you are sampling time twice, in two different instructions, and therefore the results would be wrong if you cross the boundary from one second to the next one between those two samples.

Reply:
OK, I understand the issue with my solution..  I took a different approach and collect microtime as a string, then split it on the decimal..  I think it actually makes the code easier to understand...

Reply:
I think so as well 👍